rcl/src/rcl/validate_security_context_name.c

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rcl/validate_security_context_name.h"
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <rcutils/snprintf.h>
+
+#include "rmw/validate_namespace.h"
+
+#include "rcl/error_handling.h"
+
+#include "./common.h"
+
+rcl_ret_t
+rcl_validate_security_context_name(
+  const char * security_context,
+  int * validation_result,
+  size_t * invalid_index)
+{
+  if (!security_context) {
+    return RCL_RET_INVALID_ARGUMENT;
+  }
+  return rmw_validate_namespace_with_size(
```

Comment:
@ivanpauno shouldn't this be `rcl_validate_security_context_name_with_size`?

Reply:
See https://github.com/ros2/rcl/pull/515/commits/e3d5ebdf6094274f7f74d752ecce3fbadec84150.